### PR TITLE
[docs] update `handle` docs to clarify when it runs

### PR DESCRIPTION
**Pull request by benmccann**
Closes https://github.com/sveltejs/kit/issues/2194

### CODE REVIEW

File: documentation/docs/04-hooks.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ An optional `src/hooks.js` (or `src/hooks.ts`, or `src/hooks/index.js`) file exp
 
 ### handle
 
-This function runs on every request, for both pages and endpoints, and determines the response. It receives the `request` object and a function called `resolve`, which invokes SvelteKit's router and generates a response accordingly. This allows you to modify response headers or bodies, or bypass SvelteKit entirely (for implementing endpoints programmatically, for example).
+This function runs on each endpoint request and page rendering, and determines the response. It receives the `request` object and a function called `resolve`, which invokes SvelteKit's router and generates a response accordingly. This allows you to modify response headers or bodies, or bypass SvelteKit entirely (for implementing endpoints programmatically, for example).
```
Comment (Rich-Harris) [inline documentation/docs/04-hooks.md:11]:
```suggestion
This function runs every time SvelteKit responds to a request for a page or endpoint — whether that happens while the app is running, or during [prerendering](#ssr-and-javascript-prerender) — and determines the response. It receives the `request` object and a function called `resolve`, which invokes SvelteKit's router and generates a response accordingly. This allows you to modify response headers or bodies, or bypass SvelteKit entirely (for implementing endpoints programmatically, for example).

> Requests for static assets — which includes pages that were already prerendered — are _not_ handled by SvelteKit.
```

File: documentation/docs/04-hooks.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ export async function handle({ request, resolve }) {
 }
 ```
 
+> `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page. If you would like to run some functionality on each request, see the documentation for your [adapter](#adapters).
+
```
Comment (Rich-Harris) [inline documentation/docs/04-hooks.md:64]:
```suggestion
```

**Review by Rich-Harris — CHANGES_REQUESTED** [review]
I think it probably adds more confusion than clarity to say 'see the documentation for your adapter' when none of the adapters (at least the official ones) let you intercept requests for static assets, and most of them couldn't anyway. I reckon it's easier to just say Kit doesn't handle static assets

File: documentation/docs/04-hooks.md (hunk 2 above)
Comment (benmccann) [reply documentation/docs/04-hooks.md:66]: I think we still need the first sentence here. It's not at all obvious from the text above that `handle` doesn't run on prerendered pages. I'm fine with dropping the second sentence though

File: documentation/docs/04-hooks.md [hunk 3]
```diff
@@ -60,6 +62,8 @@ export async function handle({ request, resolve }) {
 }
 ```
 
+> `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page. If you would like to run some functionality on each request, see the documentation for your [adapter](#adapters).
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:65]:
```suggestion
> `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page.
```

File: documentation/docs/04-hooks.md (hunk 2 above)
Comment (benmccann) [reply documentation/docs/04-hooks.md]: Oh, nevermind. I missed that you added a callout above

File: documentation/docs/04-hooks.md [hunk 4]
```diff
@@ -60,6 +62,8 @@ export async function handle({ request, resolve }) {
 }
 ```
 
+> `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page.
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:65]:
```suggestion
```

File: documentation/docs/04-hooks.md [hunk 5]
```diff
@@ -60,6 +62,7 @@ export async function handle({ request, resolve }) {
 }
 ```
 
+
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:65]:
```suggestion
```

**Pull request merged**